app/controllers/users_controller.rb

```diff
@@ -214,7 +222,11 @@ def parameter_filter_context
   end
 
   def verify_active_session
-    if !request.post? && params[:status].blank? && User.unscoped.exists?(session[:user].presence)
+    if User.unscoped.find_by_id(session[:user].presence)&.disabled
+      backup_session_content { reset_session }
+      return
+    end
```

Comment:
this `if` is covered by the enabled scope now, right? We won't reset the session, but I don't believe it matters. It's worth testing, but I believe we can leave the session untouched.